src/main/java/org/jboss/remoting3/remote/ClientConnectionOpenListener.java

```diff
@@ -401,7 +401,7 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
                         factoryOperator = and(ClientConnectionOpenListener.this.saslClientFactoryOperator, factoryOperator);
                         final SaslClient saslClient;
                         try {
-                            saslClient = configurationClient.createSaslClient(uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
+                            saslClient = createSaslClient(configurationClient, uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
```

Comment:
I agree this is not why the failure happens. The initial work I did was on Elytron, but maybe I was fixing to high up in the stack? When I investigated this the first time I found that the issue happened here, https://github.com/wildfly-security/wildfly-elytron/blob/7666231fa76b95c3c97eed0d16968a5f77a7de22/src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java#L54.

I'm not sure how to move this forward in a safe way, I'm afraid.